### ISSUE TITLE
Update Profile links to `View Profile` where applicable.

### ISSUE THREAD

**Issue by StevenDufresne** [WordPress/wporg-mu-plugins#268]
Originally reported in: https://meta.trac.wordpress.org/ticket/6478

When logged in to your WordPress account, user names in the Admin toolbar are linked to the edit profile page. 

@jakariaistauk337 suggests that the first 2 username links should go to `https://profiles.wordpress.org/{username}/`. There is already an "Edit Profile" link that navigates appropriately.

**Comment by dionysous** [comment]
I didn't know what components were involved at first, so I had created this Meta-Trac ticket: https://meta.trac.wordpress.org/ticket/7462 I had the same idea as @StevenDufresne and think it's important that this is discussed.

### Description
The admin bar dropdown for logged in users has redundancy in its navigation links. The intent of this ticket is to suggest an improvement to the user interface by altering one of the "Edit Profile" links to a "View Profile" link. This modification promises to improve the navigation experience by allowing users to swiftly access their public profile directly from the admin bar.

### Current Interface
- The username in the admin bar's top-right corner navigates to the Edit Profile page.
- Hovering over the username reveals the dropdown menu, which includes:
   - "Howdy, ((displayname))" redirects to the Edit Profile page.
   - "((username))" combined with "((displayname))" redirects to the Edit Profile page.
   - "Edit Profile" leads to the same Edit Profile page.
   - "Log Out" leading to the logout process.
   
**Alternative Proposed Interface Enhancement (v1)**   
![wp org-admin-bar](https://github.com/WordPress/wporg-mu-plugins/assets/12687777/2b772022-48e0-4ac1-8bdf-5dca33bea5c7)
   
### Proposed Interface Enhancement
- The username in the admin bar's top-right corner continues to navigate to the Edit Profile page.
- Hovering over the username still displays the dropdown menu, but with a more intuitive link setup:
   - "Howdy, ((displayname))" maintains its redirection to the Edit Profile page.
   - "((username))" alongside "View Profile" now directs to the View Profile page.
   - "Edit Profile" remains unchanged, leading to the Edit Profile page.
   - "Log Out" continues to facilitate the logout process.

### Challenges
- On wordpress.org, there is always a publicly visible profile. There, the "View Profile" link should always be present.
   - For example, I often use my profile to check my photo uploads in the directory. Since all three links currently lead to "Edit Profile," one must navigate further from there, which is cumbersome.
- Changing the dropdown menu could initially lead to slight confusion among the WordPress community. However, this can be prevented with clear link labels.

### Affected code
https://github.com/WordPress/wporg-mu-plugins/blob/a14fdd00695ba58dbc5b71a908780d9e7b92f537/mu-plugins/blocks/global-header-footer/admin-bar.php#L25-L29 

### Reference
- This issue is related to to this core ticket https://core.trac.wordpress.org/ticket/60489
- This issue is related to to this meta ticket https://meta.trac.wordpress.org/ticket/7462

**Comment by dionysous** [comment]
I would like to agree with @StevenDufresne and suggest changing the first 2 links:

**Alternative Proposed Interface Enhancement (v2)**
![wp org-admin-bar2](https://github.com/WordPress/wporg-mu-plugins/assets/12687777/9b1168b8-f594-4ede-83fb-04d5fcc579a8)

**Comment by StevenDufresne** [comment]
That does make sense to me, although I feel like that collapsed menu should display `Howdy, {username}` and keep the display name above view profile. 

I'm basing that on the fact that the display name isn't a clear indication of what account your are accessing, so seeing it collapsed will be beneficial in the future for other wp.org functions (password resets, etc...) and when you arrive on your profile page, you immediately see your display name as the largest landmark next to your image which should match what you see in the dropdown, so there's a flow connection there as well.

It isn't a strong opinion though, i think either change is an improvement we could make for wp.org.

**Comment by dionysous** [comment]
I agree with you. It would just be important to me to show both the _user name_ and the _display name_, as is currently the case. No matter how it is displayed, there is no question in my mind that we need _View Profile_ linking and it would simplify navigation enormously.

**Alternative Proposed Interface Enhancement (v3):**
![wp org-admin-bar3](https://github.com/WordPress/wporg-mu-plugins/assets/12687777/36cd7c02-2025-4bf0-bc8c-ead07a0edf02)

**Comment by dd32** [comment]
These items are default core objects, not customised for WordPress.org at present.

This has changed slightly since this issue was opened, as it appears to have changed in core. https://github.com/WordPress/wordpress-develop/commit/eb33c60eaa0111bac7d98a84fd0cf067cd9c8571 https://github.com/WordPress/wordpress-develop/commit/44c41a52cf653a5136fc43c9f8305ec1f5d17b1f

There's no longer a separate `Edit Profile` link, that's merged with the Avatar.

See the PR below for a change that reflects the above suggestions.

**Comment by dd32** [comment]
>  I feel like that collapsed menu should display Howdy, {username} and keep the display name above view profile.
>
> I'm basing that on the fact that the display name isn't a clear indication of what account your are accessing, so seeing it collapsed will be beneficial in the future for other wp.org functions

In the same manner, the username might not be clear as to which account you're using for some users. I don't personally think we should be displaying the username here, as that's less human than the chosen display name IMHO..
